### PR TITLE
Fix missing arguments to `WorkflowLinter`

### LINKED ISSUE #2808: Test failure: `test_linter_from_context`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Test fails

### Expected Behavior

Test should not fail

### Steps To Reproduce

Rerun test

### Cloud

Azure

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

```shell
See https://github.com/databrickslabs/ucx/actions/runs/11156195282/job/31008379387
```

### PR DESCRIPTION
**Pull request by JCZuurmond**
## Changes
Fix missing arguments to `WorkflowLinter`

### Linked issues

Resolves #2808

### CODE REVIEW

File: src/databricks/labs/ucx/contexts/application.py [hunk 1]
```diff
@@ -422,13 +423,14 @@ def dependency_resolver(self):
...
     @cached_property
-    def workflow_linter(self):
+    def workflow_linter(self) -> WorkflowLinter:
         return WorkflowLinter(
             self.workspace_client,
             self.dependency_resolver,
             self.path_lookup,
             TableMigrationIndex([]),  # TODO: bring back self.tables_migrator.index()
             self.directfs_access_crawler_for_paths,
+            self.used_tables_crawler_for_paths,
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/contexts/application.py:433]: @ericvergnaud : Is the right `UsedTablescrawler`?

**Comment by JCZuurmond** [comment]
Failing integration tests will be covered in other PRs, see integration test failure issues.

**Comment by JCZuurmond** [comment]
Want to reuse tests from #2788 to increase the test coverage which blocks the CI

File: src/databricks/labs/ucx/source_code/used_table.py [hunk 2]
```diff
@@ -15,15 +15,7 @@
...
-    @classmethod
-    def for_paths(cls, backend: SqlBackend, schema) -> UsedTablesCrawler:
-        return UsedTablesCrawler(backend, schema, "used_tables_in_paths")
-
-    @classmethod
-    def for_queries(cls, backend: SqlBackend, schema) -> UsedTablesCrawler:
-        return UsedTablesCrawler(backend, schema, "used_tables_in_queries")
-
-    def __init__(self, backend: SqlBackend, schema: str, table: str):
+    def __init__(self, backend: SqlBackend, schema: str, table: str) -> None:
```
Comment (ericvergnaud) [inline src/databricks/labs/ucx/source_code/used_table.py:18]: That's bad. For clarity class factory methods should always come before instance methods

File: tests/unit/contexts/test_application.py [hunk 3]
```diff
@@ -11,18 +12,28 @@
...
+        "notebook_loader",
+        "folder_loader",
+        "workflow_linter",
+        "used_tables_crawler_for_paths",
+        "used_tables_crawler_for_queries",
+    ],
 )
-def test_global_context_attributes_not_none(attribute: str):
+def test_global_context_attributes_not_none(attribute: str) -> None:
     """Attributes should be not None"""
```
Comment (ericvergnaud) [inline tests/unit/contexts/test_application.py:28]: should not be None ?

**Review by ericvergnaud — CHANGES_REQUESTED** [review]

File: tests/unit/contexts/test_application.py (hunk 3 above)
Comment (JCZuurmond) [reply tests/unit/contexts/test_application.py:28]: We add `-> None` to tests to trigger the linter. Actually not necessarily in this case as attribute has a type hint, but more consistent with other tests

**Pull request merged**